lib/open_uri.ml

```diff
@@ -15,36 +15,43 @@
  *
  *)
 
+open Xapi_stdext_pervasives.Pervasiveext
+
 let handle_socket f s =
```

Comment:
why not
```ocaml
try:
   Unix.connect s sockaddr;
   let result = f s in
   Unix.close s;
   result
with e ->
   Backtrace.is_important e;
   Unix.close s;
   raise e
```
it is less polished but will replace all occurrences of the finally with `handle_socket f s`. I don't thing we are using the `handle_socket` in places in which we are not calling `Unix.connect`

Reply:
This code introduce the risk of closing the fd twice.
```
try
  ...
  Unix.close s (* If exception occurs here *)
with e ->
  Unix.close s
```
So for the best idea to manage resource is using `finally`

Reply:
You convinced me :)